src/transport/msg_providers/data_bus.py

```diff
@@ -0,0 +1,281 @@
+
+    def _create_message(self, parsed_data: tuple, guardian: str) -> dict:
+        block_number, app = parsed_data
+        return PingMessageDataBus(
+            type=MessageType.PING,
+            blockNumber=block_number,
+            guardianAddress=guardian,
+        )
+
+
+# event MessagePauseV2(address indexed guardianAddress, (bytes32 depositRoot, uint256 nonce, uint256 blockNumber, bytes32 blockHash,
+# bytes signature, uint256 stakingModuleId, (bytes32 version) app) data)",
+
+
+class PauseV2Parser(LogParser):
+    def __init__(self, w3: Web3):
+        super().__init__(w3, PAUSE_V2_DATA_SCHEMA)
+
+    def _create_message(self, parsed_data: tuple, guardian: str) -> dict:
+        deposit_root, nonce, block_number, block_hash, signature, staking_module_id, app = parsed_data
+        r, _vs = signature_to_r_vs(signature)
+        return PauseMessage(
+            type=MessageType.PAUSE,
+            blockNumber=block_number,
+            guardianAddress=guardian,
+            stakingModuleId=staking_module_id,
+            signature={
+                'r': r,
+                '_vs': _vs,
+            },
+        )
+
+
+# event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, bytes signature, (bytes32 version) app) data)",
+
+
+class PauseV3Parser(LogParser):
+    def __init__(self, w3: Web3):
+        super().__init__(w3, PAUSE_V3_DATA_SCHEMA)
+
+    def _create_message(self, parsed_data: tuple, guardian: str) -> dict:
+        block_number, signature, app = parsed_data
+        r, _vs = signature_to_r_vs(signature)
+        return PauseMessage(
+            type=MessageType.PAUSE,
+            blockNumber=block_number,
+            guardianAddress=guardian,
+            signature={
+                'r': r,
+                '_vs': _vs,
+            },
+        )
+
+
+class DataBusSinks(str, Enum):
+    DEPOSIT_V1 = f'MessageDepositV1(address,{DEPOSIT_V1_DATA_SCHEMA})'
+    PAUSE_V2 = f'MessagePauseV2(address,{PAUSE_V2_DATA_SCHEMA})'
+    PAUSE_V3 = f'MessagePauseV3(address,{PAUSE_V3_DATA_SCHEMA})'
+    PING_V1 = f'MessagePingV1(address,{PING_V1_DATA_SCHEMA})'
+    UNVET_V1 = f'MessageUnvetV1(address,{UNVET_V1_DATA_SCHEMA})'
+
+
+def _construct_parsers(w3: Web3, sinks: List[DataBusSinks]) -> List[LogParser]:
+    parser_mapping = {
+        DataBusSinks.DEPOSIT_V1: DepositParser,
+        DataBusSinks.PAUSE_V2: PauseV2Parser,
+        DataBusSinks.PAUSE_V3: PauseV3Parser,
+        DataBusSinks.PING_V1: PingParser,
+        DataBusSinks.UNVET_V1: UnvetParser,
+    }
+
+    parsers = []
+    for sink in sinks:
+        parser_class = parser_mapping.get(sink)
+        if parser_class:
+            parsers.append(parser_class(w3))
+        else:
+            raise ValueError(f'Invalid sink in Data Bus sinks: {sink}')
+
+    return parsers
+
+
+class DataBusProvider(BaseMessageProvider):
+    def __init__(self, w3: Web3, message_schema: Schema, sinks: [DataBusSinks]):
+        super().__init__('', message_schema)
```

Comment:
let's create w3 instance inside constructor

Reply:
It will make difficult to inject mock instances for the unit tests